aiosmtpd/controller.py

```diff
@@ -298,38 +398,27 @@ def _trigger_server(self):
         """
         # At this point, if self.hostname is Falsy, it most likely is "" (bind to all
         # addresses). In such case, it should be safe to connect to localhost)
-        hostname = self.hostname or get_localhost()
+        hostname = self.hostname or self._localhost
```

Comment:
Rather than calling `get_localhost()` again (and _possibly_ get a different localhost IP), we call that function once and cache it.